### PR TITLE
set DSS_FAKE_504_PROBABILITY as a header, not a cookie.

### PR DESCRIPTION
**Pull request by ttung**
Connects to #485

### CODE REVIEW

File: chalice/app.py [hunk 1]
```diff
@@ -121,13 +121,10 @@ def dispatch(*args, **kwargs):
         )
 
         def maybe_fake_504() -> typing.Optional[chalice.Response]:
-            cookies = SimpleCookie(app.current_request.headers.get("Cookie", ""))
-            fake_504_probabliity_morsel = cookies.get("DSS_FAKE_504_PROBABILITY")
-            if fake_504_probabliity_morsel is None:
-                return None
+            fake_504_probabliity_str = app.current_request.headers.get("DSS_FAKE_504_PROBABILITY", "0.0")
```
Comment (kislyuk) [inline chalice/app.py]: Typo in `probabliity`

File: chalice/app.py [hunk 2]
```diff
@@ -121,13 +121,10 @@ def dispatch(*args, **kwargs):
         )
 
         def maybe_fake_504() -> typing.Optional[chalice.Response]:
-            cookies = SimpleCookie(app.current_request.headers.get("Cookie", ""))
```
Comment (kislyuk) [inline chalice/app.py]: Please remove the SimpleCookie import now that it's unused.

Comment (ttung) [reply chalice/app.py]: Done.

File: chalice/app.py (hunk 1 above)
Comment (ttung) [reply chalice/app.py]: Fixed.

**Pull request merged**